tools/clang/unittests/HLSL/OptimizerTest.cpp

```diff
@@ -1227,22 +1236,32 @@ void CompareResources(const vector<unique_ptr<ResourceType>> &original,
     auto const &optimizedRes = optimized.at(i);
     VERIFY_ARE_EQUAL(originalRes->GetClass(), optimizedRes->GetClass());
     VERIFY_ARE_EQUAL(originalRes->GetGlobalName(), optimizedRes->GetGlobalName());
-    VERIFY_ARE_EQUAL(originalRes->GetGlobalSymbol()->getName(),
-                     optimizedRes->GetGlobalSymbol()->getName());
-    VERIFY_ARE_EQUAL(originalRes->GetHLSLType()->getTypeID(),
+    if (originalRes->GetGlobalSymbol() != nullptr &&
+        optimizedRes->GetGlobalSymbol() != nullptr) {
+      VERIFY_ARE_EQUAL(originalRes->GetGlobalSymbol()->getName(),
+                       optimizedRes->GetGlobalSymbol()->getName());
+    }
+    if (originalRes->GetHLSLType() != nullptr &&
+            optimizedRes->GetHLSLType() != nullptr) {
+        VERIFY_ARE_EQUAL(originalRes->GetHLSLType()->getTypeID(),
                      optimizedRes->GetHLSLType()->getTypeID());
-    if (originalRes->GetHLSLType()->getTypeID() == Type::TypeID::StructTyID) {
-      VERIFY_ARE_EQUAL(originalRes->GetHLSLType()->getStructName(),
-                       optimizedRes->GetHLSLType()->getStructName());
+      if (originalRes->GetHLSLType()->getTypeID() == Type::PointerTyID) {
+        auto originalPointedType = originalRes->GetHLSLType()->getArrayElementType();
+        auto optimizedPointedType = optimizedRes->GetHLSLType()->getArrayElementType();
+        VERIFY_ARE_EQUAL(originalPointedType->getTypeID(),
+                           optimizedPointedType->getTypeID());
+        if (optimizedPointedType->getTypeID() == Type::StructTyID) {
+          VERIFY_ARE_EQUAL(originalPointedType->getStructName(),
+                           optimizedPointedType->getStructName());
+        }
+      }
     }
     VERIFY_ARE_EQUAL(originalRes->GetID(), optimizedRes->GetID());
     VERIFY_ARE_EQUAL(originalRes->GetKind(), optimizedRes->GetKind());
     VERIFY_ARE_EQUAL(originalRes->GetLowerBound(), optimizedRes->GetLowerBound());
     VERIFY_ARE_EQUAL(originalRes->GetRangeSize(), optimizedRes->GetRangeSize());
-    VERIFY_ARE_EQUAL(originalRes->GetResBindPrefix(), optimizedRes->GetResBindPrefix());
-    VERIFY_ARE_EQUAL(originalRes->GetResClassName(), optimizedRes->GetResClassName());
-    VERIFY_ARE_EQUAL(originalRes->GetResDimName(), optimizedRes->GetResDimName());
-    VERIFY_ARE_EQUAL(originalRes->GetResIDPrefix(), optimizedRes->GetResIDPrefix());
+    VERIFY_ARE_EQUAL_STR(originalRes->GetResBindPrefix(), optimizedRes->GetResBindPrefix());
+    VERIFY_ARE_EQUAL_STR(originalRes->GetResIDPrefix(), optimizedRes->GetResIDPrefix());
```

Comment:
These are just the prefix character decoded from the Class, so comparing `GetResBindPrefix` and `GetResIDPrefix` are also unnecessary.  Basically, if Class was wrong, there would be something very fundamentally wrong with which list the resource is in for the DxilModule.  By which I mean we would have much bigger problems, and expect a whole lot of things to break.